server/controllers/project.controller/getProjectsForUser.js

```diff
@@ -10,25 +10,44 @@ import { toApi as toApiProjectObject } from '../../domain-objects/Project';
 const createCoreHandler = (mapProjectsToResponse) => async (req, res) => {
   try {
     const { username } = req.params;
+    const currentUser = req.user?._id || '';
+
     if (!username) {
       res.status(422).json({ message: 'Username not provided' });
       return;
     }
+
     const user = await User.findByUsername(username);
     if (!user) {
       res
         .status(404)
         .json({ message: 'User with that username does not exist.' });
       return;
     }
+
     const projects = await Project.find({ user: user._id })
       .sort('-createdAt')
-      .select('name files id createdAt updatedAt')
+      .select('name files id createdAt updatedAt visibility')
```

Comment:
Now that I'm looking at this, the way that we have it currently set up is kind of silly because in the `apiGetProjectsForUser` case we could just have `'name id'` here, rather than returning the whole object from the DB and mapping it 🤔

Reply:
I believe we're using each property within `SketchList.jsx` to render the data for each row? 🤔

Reply:
Yes.  I'm thinking to myself about how to manage the two versions of the function.  Currently the difference comes from the `mapProjectsToResponse` argument.  But now my gears are turning and I'm realizing that instead we could have a `fields` argument where we pass either `'name id'` or `'name files id createdAt updatedAt visibility'` as the argument.  Just something I'm thinking about.  FYI before I worked on this file we had two completely separate functions which were *almost* identical but I loathe repetitive code so I made it a higher order function #2320.